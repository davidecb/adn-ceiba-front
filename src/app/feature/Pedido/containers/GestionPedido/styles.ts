import styled from 'styled-components';

export const DivContainer = styled.div`
  display: flex;
  flex-wrap: wrap;
  width: 100%;
  @media (max-width: 900px) {
    flex-direction: column-reverse;
  }
`;

export const DivRow = styled.div`
  min-height: 1px;
`;
