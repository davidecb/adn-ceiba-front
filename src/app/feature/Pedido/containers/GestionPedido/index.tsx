import * as PropTypes from 'prop-types';
import * as React from 'react';
import { CaracteristicasPedido } from '../../components/CaracteristicasPedido';
import { DivContainer } from './styles';
import { ListarPedido } from '../../components/ListarPedido';
import { Pedido } from '../../models/Pedido';
import { useEffect } from 'react';


interface GestionPedidoProps {
  pedido: Pedido;
  listarPedido: (idpedido: number) => void;
  finalizarPedido: (pedido: Pedido) => void;
  eliminarProductoSolicitado: (idProductoSolicitado: number) => void;
}

export const GestionPedido: React.FC<GestionPedidoProps> = ({
  pedido,
  listarPedido,
  finalizarPedido,
  eliminarProductoSolicitado,
}) => {
  useEffect(() => {
    listarPedido(pedido.id);
  }, [listarPedido]);

  return (
    <DivContainer>
          {(!pedido || pedido.costo === 0) && <h2>Aun no tienes productos en el carrito</h2>} 
          {(pedido && pedido.costo !== 0) && 
            <ListarPedido 
              pedido={pedido} 
              eliminarProductoSolicitado={
                (idProductoSolicitado: number) => eliminarProductoSolicitado(idProductoSolicitado)
              } 
            />}    
          {(pedido && pedido.costo !== 0) && 
            <CaracteristicasPedido 
              pedido={pedido} 
              onClickFinalizarPedido={finalizarPedido} 
            />}
    </DivContainer>
  );
};

GestionPedido.propTypes = {
  pedido: PropTypes.any.isRequired,
  listarPedido: PropTypes.func.isRequired,
  finalizarPedido: PropTypes.func.isRequired,
  eliminarProductoSolicitado: PropTypes.func.isRequired,
};
