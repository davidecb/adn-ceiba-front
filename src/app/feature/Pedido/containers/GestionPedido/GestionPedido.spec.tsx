import * as React from 'react';
import { RenderResult, render } from '@testing-library/react';
import { SinonStub, stub } from 'sinon';
import { GestionPedido } from './../GestionPedido';

describe('GestionPedido Test', () => {
  let componentWrapper: RenderResult;
  let componentProps: React.ComponentProps<typeof GestionPedido> & {
    listarPedido: SinonStub;
    finalizarPedido: SinonStub;
    eliminarProductoSolicitado: SinonStub;
  };

  beforeEach(() => {
    componentProps = {
      pedido: {
        id: 1,
        numeroPedido: '123456789',
        direccion: '',
        cliente: '',
        estado: '',
        costo: 10000,
        tiempo: 30,
        productosSolicitados:[]
      },
      listarPedido: stub(),
      finalizarPedido: stub(),
      eliminarProductoSolicitado: stub(),
    };
    componentWrapper = render(<GestionPedido {...componentProps} />);
  });

  it('should match snapshot', () => {
    expect(componentWrapper.container).toMatchSnapshot();
  });
});

export {};
