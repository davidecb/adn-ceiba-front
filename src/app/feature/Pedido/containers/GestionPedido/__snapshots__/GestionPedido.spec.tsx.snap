// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GestionPedido Test should match snapshot 1`] = `
<div>
  <div
    class="sc-fzozJi jRkCqE"
  >
    <div
      class="sc-fzpans jeismr"
    />
    <div
      class="sc-AxjAm dlYdrF"
    >
      <div>
        <b>
          numero de pedido:
        </b>
         
        123456789
      </div>
      <div>
        <b>
          cantidad productos:
        </b>
         
        0
      </div>
      <div>
        <b>
          costo total:
        </b>
         $
        10000
      </div>
      <div>
        <b>
          tiempo total:
        </b>
         
        30
         min.
      </div>
      <button
        class="sc-AxiKw leNlab"
      >
        <span>
          Realizar Pedido
        </span>
      </button>
    </div>
  </div>
</div>
`;
